plan/exhaust_physical_plans.go

```diff
@@ -541,33 +545,38 @@ func (la *LogicalAggregation) getStreamAggs(prop *requiredProp) []PhysicalPlan {
 	if len(la.groupByCols) != len(la.GroupByItems) {
 		return nil
 	}
-	streamAggs := make([]PhysicalPlan, 0, len(la.possibleProperties)*2)
-	for _, cols := range la.possibleProperties {
-		_, keys := getPermutation(cols, la.groupByCols)
-		if len(keys) != len(la.groupByCols) {
+
+	streamAggs := make([]PhysicalPlan, 0, len(la.possibleProperties)*(len(wholeTaskTypes)-1))
+	childProp := &requiredProp{
+		desc:        prop.desc,
+		expectedCnt: math.Max(prop.expectedCnt*la.inputCount/la.stats.count, prop.expectedCnt),
+	}
+
+	for _, possibleChildProperty := range la.possibleProperties {
+		sortColOffsets := getMaxSortPrefix(possibleChildProperty, la.groupByCols)
+		if len(sortColOffsets) != len(la.groupByCols) {
+			continue
+		}
+
+		childProp.cols = possibleChildProperty[:len(sortColOffsets)]
+		if !prop.isPrefix(childProp) {
 			continue
 		}
-		for _, tp := range wholeTaskTypes {
+
+		for _, taskTp := range wholeTaskTypes {
```

Comment:
Can we define a new constant slice that contains copSingleReadTaskType and rootTaskType? So we could save the following if.